src/collections/Menu/index.d.ts

```diff
@@ -154,6 +154,13 @@ interface MenuItemProps extends ReactMouseEvents<HTMLElement> {
   /** Internal name of the MenuItem. */
   name?: string;
 
+  /**
+   * Called after user's click.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: (event: React.MouseEvent<HTMLElement>, data?:MenuItemProps) => void;
```

Comment:
`data` will be [always](https://github.com/shanehyde/Semantic-UI-React/blob/647986443152232a9793039b8645779f90832dff/src/collections/Menu/MenuItem.js#L90) passed.

```diff
-onClick?: (event: React.MouseEvent<HTMLElement>, data?:MenuItemProps) => void;
+onClick?: (event: React.MouseEvent<HTMLElement>, data: MenuItemProps) => void;
```